app/controllers/reactions_controller.rb

```diff
@@ -4,37 +4,49 @@ class ReactionsController < ApplicationController
 
   def index
     skip_authorization
+
     if params[:article_id]
       id = params[:article_id]
-      reactions = if session_current_user_id.present?
-                    Reaction.where(reactable_id: id,
-                                   reactable_type: "Article",
-                                   user_id: session_current_user_id).
-                      where("points > ?", 0)
+
+      reactions = if session_current_user_id
```

Comment:
Rather than doing this check later in the action, can we move this check to the very top as a guard clause and then skip all of the processing that we do below if we dont have it. I noticed we also check AFTER loading a bunch of comments.

Reply:
Mmm I'm not sure we can, we load the comments because we need to compute `positive_reaction_counts` and that's needed for both signed in and non signed in users. The only thing we don't need in case the user is not signed in is the `comment_ids` array, I can move that